> I've amended these changes.

Thx. Would you mind adding the `.lib` suffix to legacy_... as well? ;)

> DMD supports older VS versions, so it should not require the existence of legacy_stdio_definitions.lib.

Are you sure that's working with pre-built druntime & Phobos? Last time I tried to use a VS2015-built LDC (incl. stdlibs) with VS 2013 (dynamic runtime),  I got an undefined reference to `..._common_printf()`; iirc, it was zlib accessing `_(v)snprintf()`, which VS 2015 declares inline as wrapper around that new common function...

--- a/src/rt/msvc.c
+++ b/src/rt/msvc.c
@@ -186,3 +186,24 @@ DECLARE_ALTERNATE_NAME (fmodf,  _msvc_fmodf);
 DECLARE_ALTERNATE_NAME (modff,  _msvc_modff);
 
 #endif // _M_IX86
+
+// LDC
+#if _MSC_VER >= 1900 // VS2015+
+    // needed for some symbols that are inline functions in the C headers
+    #pragma comment(lib, "legacy_stdio_definitions.lib")
+
+    // win32 exception handling needs VC runtime hooks
+    #ifdef _DLL
+        #ifdef _DEBUG
+            #pragma comment(lib, "vcruntimed.lib")
+        #else
+            #pragma comment(lib, "vcruntime.lib")
+        #endif
+    #else
+        #ifdef _DEBUG
+            #pragma comment(lib, "libvcruntimed.lib")
+        #else
+            #pragma comment(lib, "libvcruntime.lib")
+        #endif
+    #endif
+#endif

